include/uprotocol-cpp/uri/validator/UriValidator.h

```diff
@@ -17,23 +17,79 @@
  * KIND, either express or implied.  See the License for the
  * specific language governing permissions and limitations
  * under the License.
- * 
- * SPDX-FileType: SOURCE
- * SPDX-FileCopyrightText: 2023 General Motors GTO LLC
- * SPDX-License-Identifier: Apache-2.0
  */
 #ifndef _URI_VALIDATOR_H_
 #define _URI_VALIDATOR_H_
 
-#include <uprotocol-cpp/uri/serializer/LongUriSerializer.h>
+#include "LongUriSerializer.h"
+#include "UStatus.h"
+#include <string.h>
 
+using namespace uprotocol::utransport;
 namespace uprotocol::uri {
+/* UriValidator class to validate uri*/
+    class UriValidator {
+        public:
 
-bool valid_uri(const std::string& uri) {
-    auto uri_view = LongUriSerializer::deserialize(uri);
-    return !uri_view.isEmpty();
-}
+         /**
+          * isValidUri a {@link uri} to ensure that it has at least a name for the
+          * uEntity.
+          *
+          * @param uri {@link uri} to validate.
+          * @return Returns bool containing a true or a false
+          */
+
+          static bool valid_uri(const std::string& uri);
```

Comment:
What is the purpose of this public method? Are we not validating UUri object and not string?

Reply:
@jvnaiduwin  we should wait with this PR until URI refactoring to 1.5 is complete